src/service/viewer-impl.js

```diff
@@ -171,6 +171,12 @@ export class ViewerImpl {
     /** @private {?Promise<boolean>} */
     this.isTrustedViewer_ = null;
 
+    /** @private {boolean} */
+    this.showStoryUrlInfo_ = ampdoc.getParam('showStoryUrlInfo')
```

Comment:
can we avoid extension specific APIs in viewer. should consider moving the logic to story.

Reply:
This is a viewer specific API. When embedded in a viewer, stories render a menu that allows users to access the canonical URL. This is a security requirement for the viewers on Google properties. However, some non Google viewers do not need it and would like to opt out.

When you say that we should move the logic to the story level, do you mean that `amp-story` should look for the `?showStoryUrlInfo=0|1` parameter itself, rather than relying on the `viewer-impl` logic?

Reply:
He's saying amp-story should do `ampdoc.getParam('showStoryUrlInfo') || isEmbedded`, instead of the runtime. I like that.

Reply:
That's much better, thanks!

PTAL